app/adapters/whats_app_adapter/three_sixty_dialog_inbound.rb

```diff
@@ -22,19 +22,24 @@ def on(callback, &block)
 
     def consume(organization, whats_app_message)
       @organization = organization
+      @whats_app_message = whats_app_message
 
-      @sender = initialize_sender(whats_app_message)
+      @sender = initialize_sender
       return unless @sender
 
-      @message = initialize_message(whats_app_message)
-      return unless @message
+      @text = initialize_text
 
-      @unsupported_content = initialize_unsupported_content(whats_app_message)
+      if ephemeral_data?
```

Comment:
In general I prefer the more specific wording. Ie. instead of "ephemeral" you can just call it "template message"

Reply:
It's not just a template message though. They can use the quick reply buttons to send two responses that come as part of the template, but they can simply send the text. Also, the other two words for unsubscribing and resubscribing have nothing to do with the templates.